### ISSUE TITLE
Poetry depencency resolving is too slow (bad complexity)

### ISSUE THREAD

**Issue by chongkong** [python-poetry/poetry#1710]
- **OS version and name**: Custom debian
- **Poetry version**: 0.12.17

## Issue

Dependency resolving seems to have a bad non-linear complexity, therefore resolving dependency for even a single package takes a very long time.

For example running `poetry show tfx --tree`, the frequency for printing each line slowly decreases, and it takes few hundred seconds to print them all.

**Comment by finswimmer** [comment]
Hello @chongkong,

is it also slow with the latest preview release? Could you please provide your `pyproject.toml` for testing?

Thanks a lot.

fin swimmer

**Comment by whoizit** [comment]
```
$ poetry -V
Poetry version 1.0.0
```
```
$ time poetry add wagtail
...
Resolving dependencies... (520.3s)
...
real	21m35.501s
user	0m24.711s
sys	0m3.943s
```
```
$ pipenv --version
pipenv, version 2018.11.26
```
```
$ time pipenv install wagtail
...
real	5m58.360s
user	0m47.247s
sys	0m6.238s
```

**Comment by finswimmer** [comment]
Hello @whoizit,

which OS are you using?

My result:

```
$ time poetry add wagtail
Creating virtualenv disc in /home/finswimmer/tmp/poetry/disc/.venv
Using version ^2.7 for wagtail

Updating dependencies
Resolving dependencies... (10.8s)

Writing lock file


Package operations: 21 installs, 0 updates, 0 removals

  - Installing pytz (2019.3)
  - Installing sqlparse (0.3.0)
  - Installing certifi (2019.11.28)
  - Installing chardet (3.0.4)
  - Installing django (2.2.8)
  - Installing idna (2.8)
  - Installing six (1.13.0)
  - Installing urllib3 (1.25.7)
  - Installing webencodings (0.5.1)
  - Installing beautifulsoup4 (4.6.0)
  - Installing django-modelcluster (5.0)
  - Installing django-taggit (1.2.0)
  - Installing django-treebeard (4.3)
  - Installing djangorestframework (3.11.0)
  - Installing draftjs-exporter (2.1.7)
  - Installing html5lib (1.0.1)
  - Installing pillow (6.2.1)
  - Installing requests (2.22.0)
  - Installing unidecode (1.1.1)
  - Installing willow (1.3)
  - Installing wagtail (2.7)
poetry add wagtail  16,81s user 2,53s system 57% cpu 33,712 total
```

**Comment by whoizit** [comment]
@finswimmer fresh Arch Linux with python 3.8. 
AMD. FX-4300, 16gb ram

**Comment by sdispater** [comment]
@whoizit You issue is different from the original one. Please open a new one so we can track it properly.

The performance issues when printing lines as reported by @chongkong are fixed in `1.0.0`.
